### PR TITLE
One-shot documentation update

### PR DESCRIPTION
**Pull request by ultmaster**
Update of docs on one-shot strategies as of version 2.8.

Waiting for #4835 .

### CODE REVIEW

File: nni/retiarii/oneshot/pytorch/base_lightning.py [hunk 1]
```diff
@@ -147,34 +153,40 @@ def no_default_hook(module: nn.Module, name: str, memo: dict[str, Any], mutate_k
 class BaseOneShotLightningModule(pl.LightningModule):
 
     _mutation_hooks_note = """mutation_hooks : list[MutationHook]
-        Mutation hooks are callable that inputs an Module and returns a :class:`BaseSuperNetModule`.
+        Extra mutation hooks to support customized mutation on primitives other than built-ins.
+
+        Mutation hooks are callable that inputs an Module and returns a
+        :class:`BaseSuperNetModule <nni.retiarii.oneshot.pytorch.supermodule.base.BaseSuperNetModule>`.
```
Comment (QuanluZhang) [inline nni/retiarii/oneshot/pytorch/base_lightning.py:159]: long doc to explain mutation hooks without any simple example. a simple example could be much clearer.

**Review by QuanluZhang — APPROVED** [review]

File: docs/source/nas/exploration_strategy.rst [hunk 2]
```diff
@@ -69,7 +69,25 @@ One-shot strategy
 
 One-shot NAS algorithms leverage weight sharing among models in neural architecture search space to train a supernet, and use this supernet to guide the selection of better models. This type of algorihtms greatly reduces computational resource compared to independently training each model from scratch (which we call "Multi-trial NAS").
 
-Currently, the usage of one-shot NAS strategy is a little different from multi-trial strategy. One-shot strategy is implemented with a special type of objects named *Trainer*. Following the common practice of one-shot NAS, *Trainer* trains the super-net and searches for the optimal architecture in a single run. For example,
+Starting from v2.8, the usage of one-shot strategies are much alike to multi-trial strategies. Users simply need to create a strategy and run :class:`RetiariiExperiment <nni.retiarii.experiment.pytorch.RetiariiExperiment>`. Since one-shot strategies will manipulate the training recipe, to use a one-shot strategy, the evaluator needs to be one of the :ref:`PyTorch-Lightning evaluators <lightning-evaluator>`, either built-in or customized. Example follows:
```
Comment (cruiseliu) [inline docs/source/nas/exploration_strategy.rst:72]: Try `:class:``~nni.retiarii.experiment.pytorch.RetiariiExperiment```

File: nni/retiarii/oneshot/pytorch/base_lightning.py (hunk 1 above)
Comment (matluster) [reply nni/retiarii/oneshot/pytorch/base_lightning.py:159]: I can't think any "simple" example. Maybe I can add some reference to existing hooks.

File: docs/source/nas/exploration_strategy.rst (hunk 2 above)
Comment (matluster) [reply docs/source/nas/exploration_strategy.rst]: works like a charm. :)

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**